<launch>
<!--   <arg name="modelPath" default="$(find robocol_new_arm)/arm_description/robocol_arm.xacro meshDirectory:=$(find robocol_new_arm)/meshes/visual/" /> -->
  <arg name="planner" default="ompl" />

  <!-- By default, we do not start a database (it can be large) -->
  <!-- <arg name="db" default="false" /> -->
  <!-- Allow user to specify database location -->
  <!-- <arg name="db_path" default="$(find robocol_new_arm)/default_warehouse_mongo_db" /> -->
  
  <!-- By default, we are not in debug mode -->
  <arg name="debug" default="false" />

  <!-- By default, hide joint_state_publisher's GUI MoveIt!'s "demo" mode replaces the real robot driver with the joint_state_publisher. The latter one maintains and publishes the current joint configuration of the simulated robot. It also provides a GUI to move the simulated robot around "manually". This corresponds to moving around the real robot without the use of MoveIt. -->
  <arg name="rviz_tutorial" default="false" />
  <arg name="use_gui" default="true" />
  
  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <include file="$(find robocol_new_arm)/launch/planning_context.launch">
    <arg name="load_robot_description" value="true"/>
  </include>
  
  <!-- If needed, broadcast static tf for robot root -->
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="-0.98 1.75 -2.4 0 -1.19 1.57 world robocol_link0" /> -->
  <node pkg="tf2_ros" type="static_transform_publisher" name="virtual_joint_broadcaster_1" args="0 0 0 0 0 0 world robocol_link0" />
  
  <!-- COMENTAR DESDE ACÁ SI EL BRAZO ESTÁ CONECTADO -->
  <!-- FAKE JOINTS -->
  <!-- We do not have a robot connected, so publish fake joint states -->
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
    <param name="/use_gui" value="$(arg use_gui)"/>
    <rosparam param="/source_list">[/move_group/fake_controller_joint_states]</rosparam>
  </node>
  <!-- COMENTAR HASTA ACÁ SI EL BRAZO ESTÁ CONECTADO -->
  
  <node name="joint_state_desired_publisher" pkg="topic_tools" type="relay" args="joint_states joint_states_desired" />

  <!-- Given the published joint states, publish tf for the robot links -->
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
  
  <!-- Run the main MoveIt executable without trajectory execution (we do not have controllers configured by default) -->
  <include file="$(find robocol_new_arm)/launch/move_group.launch">
    <arg name="planner" value="$(arg planner)" />
    <arg name="allow_trajectory_execution" value="true"/>
    <arg name="fake_execution" value="true"/>
    <arg name="info" value="true"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>
  
  <!-- Run Rviz -->
  <include file="$(find robocol_new_arm)/launch/moveit_rviz.launch">
    <arg name="rviz_tutorial" value="$(arg rviz_tutorial)"/>
    <arg name="debug" value="$(arg debug)"/>
  </include>

  <!-- If database loading was enabled, start mongodb as well -->
  <!-- <include file="$(find robocol_new_arm)/launch/default_warehouse_db.launch" if="$(arg db)">
    <arg name="moveit_warehouse_database_path" value="$(arg db_path)"/>
  </include> -->

</launch>